### PR TITLE
Flare Improvements

### PR DESCRIPTION
**Pull request by Firewolf34**
# Flares with Flair
ohlawd

This is a big set of fixes to bring flares up to the level of quality they ought to be at. It fixes bugs with flares where they wouldn't show up in your hand when you held them, or make a noise when lit or striking someone, etc.etc. It also adds in a new type of flare, which is a bit brighter but shorter-lasting and less common, called the **Signalling Flare**. 

The new flare, the Signalling Flare, is a brighter, halogen-colored magnesium/phosphorus flare used by Engineering for EVA emergencies and signalling. Signalling flares...
  - ...burn hotter than a normal flare, but for less overall burn time.
  - ...make a different noise than normal when lit.
  - ...are quite a bit brighter than normal flares, but the light they produce doesn't travel out as far.
  - ...are white colored with a golden stripe.
Perhaps, one day someone can add this flare to a flaregun item.

I've also modified the sprites for the safety flare, which is the much-lower-capacity flare variant that traditionally starts in our boxes. The safety flare icon is now noticeably shorter, which fits the fact that it has like 10x less fuel than a normal flare. It also has a different, more bulbous, smaller flame.

Flare Types
From left to right, Safety Flare, Standard Flare, Signalling Flare.

I also added new flare strike sound effects, created by yours truly!

1. [Strike Sound One](https://github.com/yogstation13/Yogstation/blob/b04318049d05c6cf5253f02abae424df77a0e09a/sound/items/flare_strike_1.ogg)
2. [Strike Sound Two](https://github.com/yogstation13/Yogstation/blob/b04318049d05c6cf5253f02abae424df77a0e09a/sound/items/flare_strike_2.ogg)

And I animated the in-hand item sprite, so you can finally look like a badass running around with a lit flare. The Signalling Flare does not have a custom, white sprite, I was too lazy to add that, perhaps one of you can get around to it one day.
flare-nosmoke

Full list of changes:

  - The "safety" flares that you start with now have their own icons!

  - Flares now produce heat.

  - Flares now make the proper noise & verb when you attack someone/something with them.

  - Fixes flare item not showing up in hand.

  - Adds animations for burning flares in your hand!

  - Flares description/name changes based on the flares fuel state (spent, lit, etc.)

  - Adds 2 new ignition sounds for the flares when you pull that cord!

  - Adds new item, the Signalling Flare. 

  - Adds in signal flare to various locations in the world.

  - Adds in signal flare to engineering crew starter box.

# Comparison of flare brightness levels
### Completely Dark:
Completely Dark
### Safety Flare:
Safety Flare
### Standard Flare:
Standard Flare
### Signalling Flare:
Signalling Flare
Note: the signalling flare is red in this picture, it's an old image, it's normally white.

#### Changelog

:cl:  
rscadd: Nanotrasen has equipped Engineering staff with a brand-new flare formulation, designed for EVA work.
bugfix: Flares are no longer invisible due to a space quantum anomaly when held in the hand.
soundadd: Centcom has approved a change to make flares sound louder upon ignition, to improve crew morale.
/:cl:

### CODE REVIEW

**Comment by patpol4** [comment]
Looks like a really nice detailed PR

File: code/game/objects/items/storage/toolbox.dm [hunk 1]
```diff
@@ -46,13 +46,17 @@
...
-	switch(rand(1,3))
+	switch(rand(1,5))
 		if(1)
 			new /obj/item/flashlight(src)
 		if(2)
 			new /obj/item/flashlight/glowstick(src)
 		if(3)
 			new /obj/item/flashlight/flare(src)
+		if(4)
+			new /obj/item/flashlight/flare(src)
```
Comment (Partheo) [inline code/game/objects/items/storage/toolbox.dm:57]: Is this intentional? You have the /flare item twice.

File: code/game/objects/items/devices/flashlight.dm [hunk 2]
```diff
@@ -315,20 +322,41 @@
...
+	ignition_sound = 'sound/items/flare_strike_2.ogg'
 	frng_min = 40
 	frng_max = 70
 
+/obj/item/flashlight/flare/signal
+	name = "signalling flare"
+	desc = "A specialized formulation of the standard Nanotrasen-issued flare, containing increased magnesium content. There are instructions on the side, it reads 'pull cord, make intense light'."
+	brightness_on = 5
+	flashlight_power = 2
+	item_state = "flare"
```
Comment (Partheo) [inline code/game/objects/items/devices/flashlight.dm]: I would like white inhands for the signal flare. Working on them now.

**Review by Partheo — CHANGES_REQUESTED** [review]
Sounds are good. Sprites are good, I added inhands for the signal flare now.

**Comment by Partheo** [comment]
image

You need to change the pixel_x/pixel_y var instead of the step_x/step_y one to move the objects around. That'll fix your error.

File: code/game/objects/items/storage/toolbox.dm (hunk 1 above)
Comment (Firewolf34) [reply code/game/objects/items/storage/toolbox.dm:57]: Yes I wanted to have 2 chances for flare and 1 chance for signal flare.

**Comment by Firewolf34** [comment]
> You need to change the pixel_x/pixel_y var instead of the step_x/step_y one to move the objects around. That'll fix your error.

Okay, I have edited the map files with a text editor and replaced `step_x` with `pixel_x` and `step_y` with `pixel_y`. Hopefully that fixes it!

Thanks for the in_hands bug catch

**Review by Partheo — APPROVED** [review]

**Review by patpol4 — APPROVED** [review]
Maps seem fine but i would still like ktl's second opinion on the random flares in maint, don't they spawn from maint loot spawners?

**Review by alexkar598 — APPROVED** [review]
Code looks alright

**Comment by Firewolf34** [comment]
> Maps seem fine but i would still like ktl's second opinion on the random flares in maint, don't they spawn from maint loot spawners?

@wejengin2 : Unless the loot spawners specifically look for flare sub-types to spawn, they do not. I have not added the signal flare to any of the loot spawners intentionally. As far as I know - the only way to get your hands on a Signal Flare is to:
1. Be an engineer
2. Buy it from the gifts machines (believe it or not a flare is a "gift", spessmens are weird)
3. Get it as a lucky 1/6 random chance in a red emergency toolbox

As such, I placed one in engineering in all maps, and also in maint on certain maps which were large enough to justify it, so that you can "get lucky and find the rare shiny signal type flare".

That's just my reasoning behind it, at least.

Sidenote: I actually wanted to put one in the escape pod emergency safe, but I forgot.

**Comment by patpol4** [comment]
Alright that explanation makes sense to me

**Pull request merged**